src/main/java/org/repodriller/persistence/csv/CSVFile.java

```diff
@@ -44,10 +45,34 @@ public CSVFile(String path, String name) {
 
 	public CSVFile(String path, String name, boolean append) {
 		this(verifyPath(path) + name, append);
+	}		
+	
+	public CSVFile (String path, String name , String [] header) {
+		this(verifyPath(path) + name, header);
+	}
+	
+	public CSVFile (String fileName, String [] header) {
+		this(fileName,header,false);
 	}
 
+	public CSVFile(String fileName, String[] header, boolean append ){
+		this.header = header;
+		try {
+			ps = new PrintStream(new FileOutputStream(fileName, append));
+			String headers=Arrays.toString(header);
+			headers=headers.substring(1, headers.length()-1);
+			ps.println(headers);		
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+	
 	@Override
-	public synchronized void write(Object... line) {
+	public synchronized void write(Object... line) throws CSVFileFormatException {		
+		if (header.length == 0) header = ArrayUtils.clone(new String[line.length]);			
```

Comment:
why do you need this if? It means that every time we write, we need to allocate a new array. I think it's simpler to do `if(header!=null && header.length != line.length)`.

Then, you make `header=null` in the constructors that do not receive the header